### PR TITLE
Misc Quality of Life and Bug fixes

### PR DESCRIPTION
**Pull request by Areloch**
Added handling for if preview images on image assets fails to generate, will fallback to using the full image
Added handling for double clicking or drag-n-dropping terrain assets to create them
Improved handling of field labels in variable inspector by making the stringtable be case sensitive.
Added editor settings for handling of asset double click behavior. Can now select between Edit Asset and Spawn Asset. Support is asset type dependent.
Added editor setting for auto-importing loose files when navigating to a folder. If on and the user has flagged to also enable auto-import generally, will auto import all unaffiliated loose files in as assets.
Added editor setting for default module to use when creating new assets. Updated various tooling logic so when creating a new material, if this and the 'Always Prompt Module Target' setting is off, it will fill in the target module and target asset path info based on the default module.
Fixed issue with editors that use managedData scripts where if the path didn't exist, the script file wouldn't be made.
Fixed display issue in terrain editor where if you clear the detail map, the normal/macro/orm maps would disable, but not also clear.
Fixed handling of cleared maps in terrain editor so it no longer fills empty maps in with the no image image.
Fixed handling of creating new material where it would fill in the diffuse with a no texture image as the default.
Fixed issue where canceling out of creating a module would still prompt to create the common default folders.
Fixed issue where the Select Module window couldn't be closed via the cancel or x buttons.
Based on feedback, reduced default size of the Text Pad window.
Fixed issue where the Drop At menu list wouldn't correctly display which item was marked after it was changed.
Fixed spawning shape asset handling so it uses whatever the editor's Drop At setting is.
Improved themeing of regular bitmap buttons in the editor.
Based on feedback, flipped layout of Target Module and Target Path in the Create New Asset window.
Improved handling of setting up the Target Path for when creating new assets. If a path is not set, and the user has a Default Module, it will default the path to that module.

### CODE REVIEW

File: Templates/BaseGame/game/tools/worldEditor/scripts/menuHandlers.ed.tscript [hunk 1]
```diff
@@ -1059,7 +1059,12 @@ function EditorDropTypeMenu::onSelectItem(%this, %id, %text)
    // a drop type is selected in the menu.
    EWorldEditor.dropType = getField(%this.item[%id], 2);
    
-   %this.checkRadioItem(0, (%this.getItemCount() - 1), %id);
+   for(%i=0; %i < %this.getItemCount() + 1; %i++)
```
Comment (JeffProgrammer) [inline Templates/BaseGame/game/tools/worldEditor/scripts/menuHandlers.ed.tscript]: item count + 1? Does this go over the array?

**Review by JeffProgrammer — CHANGES_REQUESTED** [review]

**Review by JeffProgrammer — APPROVED** [review]

File: Templates/BaseGame/game/tools/assetBrowser/scripts/addModuleWindow.tscript [hunk 2]
```diff
@@ -32,7 +32,7 @@ function AssetBrowser_addModuleWindow::onGainFirstResponder(%this)
 function AssetBrowser_addModuleWindow::close()
 {
    Canvas.popDialog(AssetBrowser_addModule);  
-   eval(AssetBrowser_addModuleWindow.callbackFunction);
+   //eval(AssetBrowser_addModuleWindow.callbackFunction);
```
Comment (lukaspj) [inline Templates/BaseGame/game/tools/assetBrowser/scripts/addModuleWindow.tscript]: Might as well just delete the line

File: Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/image.tscript [hunk 3]
```diff
@@ -196,22 +196,35 @@ function AssetBrowser::buildImageAssetPreview(%this, %assetDef, %previewData)
       
       %success = saveScaledImage(%assetDef.getImagePath(), %previewFilePath);
       
+      if(%success)
+      {
```
Comment (lukaspj) [inline Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/image.tscript:200]: Indentation below this line looks off

File: Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/shape.tscript [hunk 4]
```diff
@@ -318,16 +318,23 @@ function AssetBrowser::buildShapeAssetPreview(%this, %assetDef, %previewData)
...
    if(%this.selectMode)
+   {
       %previewData.doubleClickCommand = "AssetBrowser.selectAsset( AssetBrowser.selectedAsset );";
+   }
    else
+   {
+      if(EditorSettings.value("Assets/Browser/doubleClickAction", "Edit Asset") $= "Edit Asset")
       %previewData.doubleClickCommand = "AssetBrowser.editAsset( "@%assetDef@" );";
+      else
+         %previewData.doubleClickCommand = "AssetBrowser.onShapeAssetEditorDropped( "@%assetDef@" );";
```
Comment (lukaspj) [inline Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/shape.tscript:332]: Generally I think if you don't have braces it should be a single line. I'd prefer braces here

File: Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/shape.tscript [hunk 5]
```diff
@@ -340,7 +347,7 @@ function AssetBrowser::onShapeAssetEditorDropped(%this, %assetDef, %position)
    else
    {
       %pos = "0 0 0";  
-   }
+   }*/
```
Comment (lukaspj) [inline Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/shape.tscript]: Might as well delete the code

File: Templates/BaseGame/game/tools/gui/profiles.ed.tscript [hunk 6]
```diff
@@ -353,13 +353,18 @@ new GuiControlProfile( ToolsGuiButtonProfile )
...
+   fillColorNA = EditorSettings.value("Theme/tabsSELColor");
+   
    fontColor = EditorSettings.value("Theme/fieldTextColor");
    fontColorHL = EditorSettings.value("Theme/fieldTextHLColor");
    fontColorNA = EditorSettings.value("Theme/fieldTextSELColor");
    fixedExtent = false;
    justify = "center";
    canKeyFocus = false;
-	bitmapAsset = "ToolsModule:button_image";
+	//bitmapAsset = "ToolsModule:button_image";
```
Comment (lukaspj) [inline Templates/BaseGame/game/tools/gui/profiles.ed.tscript]: Delete This line instead

**Pull request merged**